modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/AddressEndpointFactory.java

```diff
@@ -115,7 +119,16 @@ public EndpointDefinition createEndpointDefinition(OMElement elem) {
         }
 
         if (address != null) {
-            endpointDefinition.setAddress(address.getAttributeValue().trim());
+            String extractedAddress = address.getAttributeValue().trim();
+            if (extractedAddress.contains(SYSTEM_VARIABLE_PREFIX)) {
+                String extractedEnvVariableKey = extractedAddress.substring(extractedAddress.lastIndexOf(":") + 1);
+                String extractedEnvVariableValue = System.getenv(extractedEnvVariableKey);
+                log.info ("Environment variable " + extractedEnvVariableKey + " replaced with " +
```

Comment:
Make this a log.debug

Reply:
Removed repeated lines and updated to log.debug in the new unified method declaration